examples/linear_model/plot_quantile_regression.py

```diff
@@ -11,100 +12,296 @@
+# case to estimate the conditional quantile (e.g. median) instead of the
+# conditional mean as in :class:`sklearn.linear_model.QuantileRegressor`.
+#
+# In the remainder of this tutorial, we will show how
+# :class:`~sklearn.linear_model.QuantileRegressor` can be used in practice and
+# give the intuitions regarding the fitting properties of the model. Finally,
+# we will compare the both :class:`~sklearn.linear_model.QuantileRegressor`
+# and :class:`~sklearn.linear_model.LinearRegression`.
+#
+# Fitting a `QuantileRegressor`
+# -----------------------------
+#
+# In this section, we want to estimate the conditional median as well as
+# a low and high quantiles fixed at 5% and 95%, respectively. Thus, we will get
+# three linear models, one for each quantile.
+#
+# We will use the quantile at 5% and 95% to find the training samples that
+# are outside of this interval.
+from sklearn.linear_model import QuantileRegressor
 
 quantiles = [0.05, 0.5, 0.95]
-models = []
+predictions = {}
+out_bounds_predictions = np.zeros_like(y_true, dtype=np.bool_)
 for quantile in quantiles:
     qr = QuantileRegressor(quantile=quantile, alpha=0)
-    qr.fit(X, y)
-    ax1.plot(x, qr.predict(X))
-    models.append(qr)
+    y_pred = qr.fit(X, y_noisy_normal).predict(X)
+    predictions[quantile] = y_pred
 
-plot_points_highlighted(x, y, models[0], models[2], ax1)
-ax1.set_xlabel("x")
-ax1.set_ylabel("y")
-ax1.set_title("Quantiles of heteroscedastic Normal distributed target")
-ax1.legend(["true mean"] + quantiles)
+    if quantile == min(quantiles):
+        out_bounds_predictions = np.logical_or(
+            out_bounds_predictions, y_pred >= y_noisy_normal
+        )
+    elif quantile == max(quantiles):
+        out_bounds_predictions = np.logical_or(
+            out_bounds_predictions, y_pred <= y_noisy_normal
+        )
 
+# %%
+# Now, we can plot the three linear models and the distinguished samples that
+# are within the 5%-95% quantiles from samples that are outside these bounds.
+plt.plot(X, y_true, color="black", linestyle="dashed", label="True mean")
 
-a = 5
-y = 10 + 0.5 * x + 10 * (rng.pareto(a, size=x.shape[0]) - 1 / (a - 1))
-ax2.plot(x, y_mean, "k--")
+for quantile, y_pred in predictions.items():
+    plt.plot(X, y_pred, label=f"Quantile: {quantile}")
+
+plt.scatter(
+    x[out_bounds_predictions],
+    y_noisy_normal[out_bounds_predictions],
+    color="black",
+    marker="+",
+    alpha=0.5,
+    label="Outside interval",
+)
+plt.scatter(
+    x[~out_bounds_predictions],
+    y_noisy_normal[~out_bounds_predictions],
+    color="black",
+    alpha=0.5,
+    label="Inside interval",
+)
 
-models = []
+plt.legend()
+plt.xlabel("x")
+plt.ylabel("y")
+_ = plt.title("Quantiles of heteroscedastic Normal distributed target")
+
+# %%
+# Since the noise is still Normally distributed, the true mean and conditional
+# median estimated are close to each other. We observe the effect of having an
```

Comment:
```suggestion
# Since the noise is still Normally distributed, in particular is symmetric, the true conditional mean
# and the true conditional median coincide. Indeed, we see that the estimated median almost
# hits the true mean. We observe the effect of having an
```